tools/release-ci/cmd/sidecar.go

```diff
@@ -100,31 +106,38 @@ func retrieveProcessReturnCode(markerFile string) (int, error) {
 	defer watcher.Close()
 
 	group := sync.WaitGroup{}
-	group.Add(1)
-	go func() {
-		defer group.Done()
-		for {
-			select {
-			case event := <-watcher.Events:
-				if event.Name == markerFile && event.Op&fsnotify.Create == fsnotify.Create {
-					break
+
+	// Only start watching file events if the file doesn't exist
+	// If the file exists, it means the main process already completed.
+	if !exists(markerFile) {
+		group.Add(1)
+		go func() {
+			defer group.Done()
+		outerLoop:
```

Comment:
Does a refactor without labels look uglier?

Reply:
No, I can do that. I was not thinking of making it pretty, but as it was, it never exited the loop